test/suites/integration/networkinterfaces_test.go

```diff
@@ -0,0 +1,116 @@
+package integration_test
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/test"
+	"github.com/aws/karpenter/pkg/apis/settings"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	awstest "github.com/aws/karpenter/pkg/test"
+)
+
+var _ = Describe("NetworkInterfaces", func() {
+	It("should create a default NetworkInterface if none specified", func() {
+		provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+			AWS: v1alpha1.AWS{
+				SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+				SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+				LaunchTemplate:        v1alpha1.LaunchTemplate{},
+			},
+		})
+		provisioner := test.Provisioner(test.ProvisionerOptions{ProviderRef: &v1alpha5.MachineTemplateRef{Name: provider.Name}})
+		pod := test.Pod()
+		env.ExpectCreated(pod, provider, provisioner)
+		env.EventuallyExpectHealthy(pod)
+		env.ExpectCreatedNodeCount("==", 1)
+		instance := env.GetInstance(pod.Spec.NodeName)
+		Expect(instance.NetworkInterfaces).To(HaveLen(1))
+		Expect(instance.NetworkInterfaces[0]).ToNot(BeNil())
+		Expect(instance.NetworkInterfaces[0].Attachment).To(HaveField("DeviceIndex", HaveValue(Equal(int64(0)))))
+		Expect(instance.NetworkInterfaces[0].Attachment).To(HaveField("NetworkCardIndex", HaveValue(Equal(int64(0)))))
+		//Expect(instance.PublicIpAddress).To(BeNil())
+	})
+	It("should use the specified NetworkInterface", func() {
+		desc := "a test network interface"
+		provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+			AWS: v1alpha1.AWS{
+				SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+				SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+				LaunchTemplate: v1alpha1.LaunchTemplate{
+					NetworkInterfaces: []*v1alpha1.NetworkInterface{
+						{
+							AssociatePublicIPAddress: aws.Bool(true),
+							Description:              aws.String(desc),
+							DeviceIndex:              aws.Int64(0),
+							NetworkCardIndex:         aws.Int64(0),
+							IPv4PrefixCount:          aws.Int64(2),
```

Comment:
since prefix delegation is only available for nitro instance types, you'll need to add a Provisioner requirement:

```
key: karpenter.k8s.aws/instance-hypervisor
operator: In
values: ["nitro"]
```

also, Is it valid to do both an ipv4 and ipv6 prefix?

Reply:
I added the requirements

> Is it valid to do both an ipv4 and ipv6 prefix?
I do not know 🤔